### PR TITLE
fix(DTFS2-5938) - Facility premium schedule + exposure date re-calculated

### PR DESCRIPTION
**Pull request by abhi-markan**
# Introduction
Once a facility has been issued and submitted to UKEF (usually as an MIA) and if the cover start date is then subsequently amended by the bank this is not taken into consideration thus invalid GCD, PS and exposure period is calculated.

This PR aims to mitigate this issue.

# Resolution
* Introduced `sendToUkef` constant, evaluates whether facility properties recalculation is necessary.

## First submission
* AIN
  * Unissued : Skip
  * Issued : Calculate and `hasBeenIssuedAndAcknowledged` set to `true`
* MIA
  * Unissued : Skip
  * Issued : Calculate and `hasBeenIssuedAndAcknowledged` set to `true`
* MIN : N/A

## Second submission
* AIN
  * Unissued : N/A
  * Issued : Calculate as `hasBeenIssuedAndAcknowledged` does not exists.
* MIA
  * Unissued : N/A
  * Issued : Calculate as `hasBeenIssuedAndAcknowledged` does not exists.
* MIN
  * Unissued : N/A
  * Issued : Calculate as `hasBeenIssuedAndAcknowledged` set to true previously but set to `false` for `MIN`.

# Miscellaneous
* Lint improvements.
* Documentation enhancement.

### CODE REVIEW

File: portal/server/routes/contract/bond/index.js [hunk 1]
```diff
@@ -38,24 +38,14 @@ const userCanAccessBond = (user, deal) => {
 
   const { status } = deal.details;
 
-  if (status === 'Ready for checker\'s approval'
+  return !(status === 'Ready for checker\'s approval'
```
Comment (IgnatG) [inline portal/server/routes/contract/bond/index.js:41]: should we add this inside an array? there are a few conditions here
i.e.
```
const array = ['Ready for checker\'s approval', 'Acknowledged', etc]
```
these items ^ could come from CONSTANTS as well
```
if (!array.includes(status)) {...}
```

File: trade-finance-manager-api/src/v1/controllers/update-issued-facilities.js [hunk 2]
```diff
@@ -21,19 +21,27 @@ const updatedIssuedFacilities = async (deal) => {
...
+     * `sendToUkef` : Facility has been sent to UKEF.
+     *
+     * Ensures tasks only done once and email only sent once for each issued
+     * facility hasBeenIssuedAndAcknowledged only set by tfm-api during this step.
+     *
+     * If MIN then set `sendToUkef` to `false` to accommodate any cover start date
+     * amendment.
     */
-    if (hasBeenIssued && !hasBeenIssuedAndAcknowledged) {
+    const sendToUkef = submissionType === CONSTANTS.DEALS.SUBMISSION_TYPE.MIN
```
Comment (IgnatG) [inline trade-finance-manager-api/src/v1/controllers/update-issued-facilities.js:40]: based on this comment
> * `sendToUkef` : Facility has been sent to UKEF.

should this be `sentToUkef`?

Comment (abhi-markan) [reply trade-finance-manager-api/src/v1/controllers/update-issued-facilities.js:40]: Good spot, have updated.

File: portal/server/routes/contract/bond/index.js (hunk 1 above)
Comment (abhi-markan) [reply portal/server/routes/contract/bond/index.js:41]: Good spot, have improved the code quality.

**Review by IgnatG — APPROVED** [review]

**Review by IgnatG — APPROVED** [review]

**Pull request merged**